app/models/notification.rb

```diff
@@ -66,10 +66,10 @@ def set_notification_recipients
 
   def set_custom_attributes
     return unless notification_blueprint # let validation catch this.
-    self.actions ||= UINotifications::URLResolver.new(
+    self.actions = UINotifications::URLResolver.new(
       subject,
       notification_blueprint.actions
-    ).actions if notification_blueprint.actions.any?
+    ).actions if notification_blueprint.actions.any? && self.actions.blank?
```

Comment:
Not a ruby developer, but multi line statements with guards are hard to parse for me.

Reply:
I'm not making the problem worse, since this will be cherrypicked I'd probably keep it as it is, I can open separate PR with refactoring if you prefer

Reply:
Sounds good.

Reply:
https://github.com/theforeman/foreman/pull/5170 addresses it